### ISSUE TITLE
🐛Docker image not outputting gif or video to artifacts locally

### ISSUE THREAD

**Issue by Fluidbyte** [qawolf/qawolf#348]
It's outputting the `log` and `html` file but really looking for the video files. I see that `xvfb` is [installed here](https://github.com/qawolf/qawolf/blob/master/Dockerfile#L8) but I don't see `rrweb`. Is this intentional or can I create a PR for this? Assuming this is the reason I'm not getting the files.

**Comment by jperl** [comment]
It should work. Where are you running this docker image?

[This is an example](https://github.com/qawolf/node-example) running qawolf in the various CIs.

**Comment by Fluidbyte** [comment]
Right now I'm just running it locally with the following:

```
docker run --rm -t --net=host -v $PWD:/app \
    -w /app/__qa__ \
    -e QAW_HEADLESS=true \
    -e QAW_SLEEP_MS=0 \
    -e QAW_TIMEOUT_MS=10000 \
    -e QAW_ARTIFACT_PATH=./artifacts \
    qawolf/qawolf:v0.7.10 \
    /bin/bash -c "rm -rf /app/__qa__/artifacts && /app/node_modules/.bin/qawolf test"
```

**Comment by jperl** [comment]
Please use the latest version `0.8.1`. `v0.8.x` [simplified](https://github.com/qawolf/qawolf/compare/v0.7.11..v0.8.1#diff-3254677a7917c6c01f55212f86c57fbf) the way the xvfb / recording works.

**Comment by Fluidbyte** [comment]
I just updated it and still no luck - just the log and html.

**Comment by jperl** [comment]
Ah I see the issue, if you set QAW_HEADLESS=true it will not record. It creates a virtual display and need to be headful to record a video.

**Comment by jperl** [comment]
To clarify about rrweb, the .html file is the rrweb player with the rrweb dom recording. However I recommend relying on the video for debugging, because right now there are various issues with rrweb. For example, it does not support css-in-js yet.

**Comment by Fluidbyte** [comment]
Got it, so since I'm running this on a remote box without Gtk is there no way to get these artifacts?

**Comment by jperl** [comment]
If you can run the docker image on that box you should be fine. Or if you can install xvfb, which is what is used to create the virtual display.

**Comment by jperl** [comment]
Also happy to [jump on a call](https://calendly.com/jperl/30min) and talk through your use case.

**Comment by Fluidbyte** [comment]
yeah, the docker image works fine with `QAW_HEADLESS`, I have xvfb installed. Getting the following without `QAW_HEADLESS`:

```
Failed to launch chrome!
    Fontconfig warning: "/etc/fonts/fonts.conf", line 100: unknown element "blank"

    (chrome:144): Gtk-WARNING **: cannot open display: 


    TROUBLESHOOTING: https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md

      at onClose (../../node_modules/puppeteer/lib/Launcher.js:348:14)
      at ChildProcess.<anonymous> (../../node_modules/puppeteer/lib/Launcher.js:338:60)
```

**Comment by jperl** [comment]
That's odd, I will run it locally and send you a command that works.

**Comment by Fluidbyte** [comment]
👍 thanks so much, absolutely loving QAWolf and your responsiveness here is amazing! I've been touting this project to all my friends and sending them screenshots of my ridiculously fast QA times 😃

**Comment by jperl** [comment]
Glad you like it and thanks for sharing it! We have a lot of improvements planned (the next release will make test creation and editing interactive), and I am always looking for suggestions.

I will get that command to you shortly.

**Comment by jperl** [comment]
Thank you for reporting this. I found the issue. The display is never created because we only create the display [on CI](https://github.com/qawolf/qawolf/blob/master/packages/screen/src/VirtualCapture.ts#L54). This  does not make sense, was a relic from the refactor. I will remove that and add better logging for when recording is not available (ex. if not on linux or xvfb not found) in the next release.

Here is a command that works for running `node-example` on my mac.

```bash
docker run --rm -w /app -v $PWD:/app -v /app/node_modules -e CIRCLECI=true -e QAW_ARTIFACT_PATH=./artifacts -e URL="http://host.docker.internal:3000" -it qawolf/qawolf:v0.8.1 /bin/bash -c "npm i && npx qawolf test" 
```

- I override `node_modules` with an empty mount ` -v /app/node_modules` after the first mount, and freshly install the node_modules with `npm i`
- `-e CIRCLECI=true` will not be necessary in the next version, it is just a workaround until this fix is released
- `URL` is specific to my app

**Comment by Fluidbyte** [comment]
@jperl that worked perfectly, thank you so much! I have both `gif` and `mp4` files which is great. Feel free to cancel our call this afternoon - I'm all set.

**Comment by jperl** [comment]
Glad it worked, and sorry there was this bug. It will be fixed in the next release with tests. I cancelled this afternoon, but feel free to schedule with me on that link if you ever have something urgent.

**Comment by jperl** [comment]
Note to self when fixing this:

- check if ffmpeg is available
- [log](https://github.com/qawolf/qawolf/blob/master/packages/screen/src/VirtualCapture.ts#L29) when recording is disabled and why (not on linux/no ffmpeg found)

**Comment by jperl** [comment]
@Fluidbyte this fix is now released in the latest version (v0.8.2)

**Comment by Fluidbyte** [comment]
@jperl 
![you rock](https://media1.giphy.com/media/wFOC9RazP97i0/giphy.gif?cid=790b7611f91b0e8c3ff8b00a21e8a69b935a47d11770dcf2&rid=giphy.gif)
